filter.js

```diff
@@ -225,20 +226,23 @@ function noMatchingSearch(printToCSS) {
 
 function filterFunctionSearchBar() {
     const findThis = filterSearchBar.value.toLowerCase().split(" ");
-    testbox.innerHTML = ""
     const hit = true;
     fullApiJson.forEach(challenge => {
-        if (findThis[0] == '') {
+        if (findThis[0] == '' || filterSearchBar.value.length <= 2) {
+            testbox.innerHTML = ""
             printAllChallenges()
         }
-        if (findThis.some(test => (test != "") && (challenge.title.toLowerCase().includes(test) || challenge.description.toLowerCase().includes(test)))) {
-            const challengeBox = createChallengeBox(challenge)
-            testbox.appendChild(challengeBox);
-        };
+        if (2 < filterSearchBar.value.length) {
```

Comment:
Great work. Together with row 231, 232 this looks like it should work. 
No comment for the loadingBar due to its own merged branch and testing. :)